packages/ckeditor5-fullscreen/src/handlers/abstracteditor.ts

```diff
@@ -122,4 +140,70 @@ export default class AbstractEditorHandler {
 			this._container = null;
 		}
 	}
+
+	/**
+	 * Stores the current state of the annotations UIs to restore it when leaving fullscreen mode.
+	 * Code coverage is provided in the commercial package repository as integration unit tests.
+	 */
+	// /* istanbul ignore next -- @preserve */
+	private _overrideAnnotationsUIs() {
+		const annotationsUIs = this._editor.plugins.get( 'AnnotationsUIs' ) as AnnotationsUIs;
+
+		this.annotationsUIsData = new Map( annotationsUIs.uisData );
+
+		// Switch to the wide sidebar.
+		const sidebarPlugin = this._editor.plugins.get( 'Sidebar' ) as Sidebar;
+
+		if ( !sidebarPlugin.container ) {
+			sidebarPlugin.setContainer(
+				this.getContainer().querySelector( '[data-ck-fullscreen="right-sidebar"]' ) as HTMLElement
+			);
+		}
+
+		const annotationsFilters = new Map<string, ( annotation: Annotation ) => boolean>();
+
+		for ( const [ uiName, data ] of [ ...this.annotationsUIsData! ] ) {
+			// Default filter is `() => true`. Only store filters that are different.
+			if ( data.filter.toString() !== '() => true' ) {
+				annotationsFilters.set( uiName, data.filter );
+			}
+		}
```

Comment:
I think we should have an API in the `Annotations` plugin for that. Anyway, for now, we can ignore it. However, I would create a follow-up task to rethink and improve it.

Reply:
https://github.com/cksource/ckeditor5-commercial/issues/7153